controllers/discovery/initDiscoveryDomainData.js

```diff
@@ -45,11 +56,21 @@ const InitDiscoveryDomainData = async (req, res) => {
     });
     let suggestedDomains = domainWithCommonFollowerResult;
 
+    let totalData = await sequelize.query(totalDataQuery, {
+      type: QueryTypes.SELECT,
+      replacements: {
+        userId
+      },
+      raw: true
+    });
+    totalData = totalData?.[0]?.total_data || 0;
+
     return res.status(200).json({
       success: true,
       message: `Fetch discovery data success`,
       suggestedDomains,
-      page: page + 1
+      page: page + 1,
+      total_page: totalData > 0 ? Math.ceil(totalData / limit) : 0
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

There is a potential division by zero error at line 73. If `limit` is 0, it will cause a division by zero error. You should add a check for `limit` before performing the division.
```diff
- total_page: totalData > 0 ? Math.ceil(totalData / limit) : 0
+ total_page: totalData > 0 && limit > 0 ? Math.ceil(totalData / limit) : 0
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->

Reply:
@n-permana please follow this recommendation